### PR TITLE
Nested Folders: Support getting of nested folder in folder service when feature flag is set 

### PR DESCRIPTION
**Pull request by papagian**
**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/api/dashboard.go [hunk 1]
```diff
@@ -394,14 +396,18 @@ func (hs *HTTPServer) postDashboard(c *models.ReqContext, cmd models.SaveDashboa
 	cmd.OrgId = c.OrgID
 	cmd.UserId = c.UserID
 	if cmd.FolderUid != "" {
-		folder, err := hs.folderService.GetFolderByUID(ctx, c.SignedInUser, c.OrgID, cmd.FolderUid)
+		ctx := appcontext.WithUser(c.Req.Context(), c.SignedInUser)
```
Comment (mildwonkey) [inline pkg/api/dashboard.go:399]: The `c.Req.Context()` will already have the signed in user set; that's handled by the [contextHandler middleware](https://github.com/grafana/grafana/blob/main/pkg/services/contexthandler/contexthandler.go#L101)

We should instead pass the entire `c.Req.Context` as the context for the folder service; that's exactly what it's there for :)

**Review by mildwonkey — CHANGES_REQUESTED** [review]
I left one comment, though I saw the behavior elsewhere too - we don't need to build a new context, the whole point is that the same context can be passed directly into the service.

File: pkg/api/folder.go [hunk 2]
```diff
@@ -221,6 +226,42 @@ func (hs *HTTPServer) toFolderDto(c *models.ReqContext, g guardian.DashboardGuar
...
+		CanEdit:   canEdit,
+		CanAdmin:  canAdmin,
+		CanDelete: canDelete,
+		CreatedBy: creator,
+		Created:   folder.Created,
+		UpdatedBy: updater,
+		Updated:   folder.Updated,
+		// Version:       folder.Version,
+		AccessControl: hs.getAccessControlMetadata(c, c.OrgID, dashboards.ScopeFoldersPrefix, folder.UID),
+	}
```
[the remark this answers is not in the document]
Comment (mildwonkey) [reply pkg/api/folder.go:262]: We don't need Version or HasACL (all folders have an acl), we do need URL.

[the remark this answers is not in the document]
Comment (papagian) [reply pkg/api/folder.go:262]: > In the new "nested folder" world, we don't need Version or HasACL (all folders have an acl); we do need to add URL back in though (there's a ticket for that already, but feel free).

not strong opinions but in the existing API (and UI) updates use [the version](https://github.com/grafana/grafana/blob/fdeefaee42d252bb1193ee60535e2fb4431ec23e/pkg/models/folders.go#L85) for preventing overrides.

> I had originally imagined that the service <-> http API communication would continue using existing models, while the folder.Folder model would be used primarily by the folder store until we removed all of the legacy (dashboard-style) folders.
I think that I'm unable to think clearly right now, but if we use the existing models for the API how we are going to provide the `ParentUID` during creates and updates?

[the remark this answers is not in the document]
Comment (mildwonkey) [reply pkg/api/folder.go:262]: It's not you @papagian, that's entirely logical. We would either need to add ParentUID to the existing models, or continue to use the folder_id field and call `GetFolderByID()` whenever a service needs the parent folder's UID. Your suggestion, adding parentUID, is *much better*, and anyway making changes to the folder *DTO* to support nested folders makes sense to me.

I've previously discussed the version field with the workgroup and (so far) we are agreed that the version field is not needed with folders. The only necessary gate around changing folders is the authnz.

**Comment by mildwonkey** [comment]
I might not be able to dismiss my 'requested changes' until this PR is marked ready for review. My request can be dismissed, in case I'm not working when this is ready to merge. Don't block on me!

**Pull request closed without merging**